src/context/WalletProvider/NewWalletViews/sections/SavedWalletsSection.tsx

```diff
@@ -0,0 +1,182 @@
+      const Vault = await import('@shapeshiftoss/hdwallet-native-vault').then(m => m.Vault)
+
+      const storedWallets: VaultInfo[] = await Vault.list().then(vaultIds =>
+        Promise.all(
+          vaultIds.map(async id => {
+            const meta = await Vault.meta(id)
+            const name = String(meta?.get('name') ?? id)
+            return { id, name }
+          }),
+        ),
+      )
+
+      return storedWallets
+    },
+    refetchOnMount: true,
+  })
+
+  const handleWalletSelect = useCallback(
+    async (wallet: VaultInfo) => {
+      // Ensure we're at the correct route when selecting a saved wallet
+      dispatch({ type: WalletActions.SET_INITIAL_ROUTE, payload: '/native/enter-password' })
+      // Ensure the wallet is visually selected in the left wallets list
+      onWalletSelect(wallet.id, '/native/enter-password')
+
+      const adapter = await getAdapter(KeyManager.Native)
+      const deviceId = wallet.id
+      if (adapter) {
+        const { name, icon } = NativeConfig
+        try {
+          dispatch({
+            type: WalletActions.SET_NATIVE_PENDING_DEVICE_ID,
+            payload: deviceId,
+          })
+
+          const walletInstance = await adapter.pairDevice(deviceId)
+          if (!(await walletInstance?.isInitialized())) {
+            await walletInstance?.initialize()
+          } else {
+            dispatch({
+              type: WalletActions.SET_WALLET,
+              payload: {
+                wallet: walletInstance,
+                name,
+                icon,
+                deviceId,
+                meta: { label: wallet.name },
+                connectedType: KeyManager.Native,
+              },
+            })
+            dispatch({
+              type: WalletActions.SET_IS_CONNECTED,
+              payload: true,
+            })
+            dispatch({ type: WalletActions.RESET_NATIVE_PENDING_DEVICE_ID })
+            dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: false })
+          }
+
+          localWallet.setLocalWallet({ type: KeyManager.Native, deviceId })
+          localWallet.setLocalNativeWalletName(wallet.name)
+        } catch (e) {
+          console.error(e)
+        }
+      }
+    },
+    [dispatch, getAdapter, localWallet, onWalletSelect],
+  )
+
+  const handleAddNewWalletClick = useCallback(() => {
+    history.push(NativeWalletRoutes.Connect)
+  }, [history])
+
+  if (!nativeVaultsQuery.data?.length) return null
+
+  return (
+    <>
+      <Text
+        fontSize='xl'
+        fontWeight='semibold'
+        translation='walletProvider.shapeShift.onboarding.shapeshiftNative'
+      />
+      <Stack spacing={2} my={6}>
+        {nativeVaultsQuery.data.map(wallet => (
+          <WalletCard
+            wallet={wallet}
+            onClick={handleWalletSelect}
+            isSelected={selectedWalletId === wallet.id}
+          />
+        ))}
+        <Box
+          as={Button}
+          variant='ghost'
+          px={4}
+          ml={-4}
```

Comment:
q: Maybe we should wrap those buttons and wallet buttons into a container and ad this negative margin to the container, because we will end up having weird spacing differences in the future if someone forget to add those styles

Reply:
Also going to tackle in follow-up PR 🙏🏽